x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/AbstractScriptFieldScript.java

```diff
@@ -45,13 +49,25 @@
         );
     }
 
+    private static final Map<String, Function<Object, Object>> PARAMS_FUNCTIONS = Map.of(
+        "_source",
+        value -> ((SourceLookup) value).loadSourceIfNeeded()
+    );
+
     private final Map<String, Object> params;
     private final LeafSearchLookup leafSearchLookup;
 
     public AbstractScriptFieldScript(Map<String, Object> params, SearchLookup searchLookup, LeafReaderContext ctx) {
         this.leafSearchLookup = searchLookup.getLeafSearchLookup(ctx);
-        // TODO how do other scripts get stored fields exposed? Through asMap? I don't see any getters for them.
-        this.params = params;
+        params = new HashMap<>(params);
+        for (Map.Entry<String, Object> entry : leafSearchLookup.asMap().entrySet()) {
+            // accessing doc_values through through params._doc and params.doc is deprecated in all existing script contexts,
+            // it makes little sense to expose it deprecated for runtime fields
+            if (entry.getKey().equals("_doc") == false && entry.getKey().equals("doc") == false) {
+                params.put(entry.getKey(), entry.getValue());
```

Comment:
Do you think it'd be more clear to:
```
params.put("_source", leafSearchLookup.searchLookup());
params.put("_fields", leafSearchLookup.fields());
```
?

Reply:
possibly too clear :D

Reply:
I would not rely on `asMap()`, but be explicit as Nik suggests.

Reply:
agreed, I already made this change